### ISSUE TITLE
Generic instance variable can have anything assigned to it

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#3872]
**To Reproduce**
```python
#pyright: strict

from typing import Generic, TypeVar

T = TypeVar("T")

class Foo(Generic[T]):
    def __init__(self, value: T) -> None:
        super().__init__()
        self.value: T = value

    def set_value(self, value: int):
        self.value = value # no error is reported here
```

**Expected behavior**
A typing mismatch error on the last line, preventing values that are not of type `T@Foo` from being assigned to `self.value`

**VS Code extension or command-line**
Extension and cli version 1.1.268

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.269, which I just published. It will also be included in a future version of pylance.
